presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -581,12 +587,12 @@ public Boolean process(Node node, @Nullable Void context)
 
     private boolean hasOrderByReferencesToOutputColumns(Node node)
     {
-        return !getReferencesToScope(node, analysis, orderByScope.get()).isEmpty();
+        return hasReferencesToScope(node, analysis, orderByScope.get());
```

Comment:
Optional: in the `process` method above there is condition:
`&& (!orderByScope.isPresent() || !hasOrderByReferencesToOutputColumns(node))`

This condition is OK, but probably better (more natural) condition would be:
`Are free column references all from GROUP BY (or outer scope)`?

This way when lambda name resolution uses Scopes too (see: https://github.com/prestodb/presto/issues/7780), then it would be possible to remove this lambda specific condition:
`&& !hasReferencesToLambdaArgument(node, analysis)) {`